cmd/api/src/database/sso_providers.go

```diff
@@ -69,3 +71,12 @@ func (s *BloodhoundDB) GetAllSSOProviders(ctx context.Context, order string, sql
 	result := query.Find(&providers)
 	return providers, CheckError(result)
 }
+
+func (s *BloodhoundDB) GetSSOProviderBySlug(ctx context.Context, slug string) (model.SSOProvider, error) {
+	var provider model.SSOProvider
+	if tx := s.db.WithContext(ctx).Raw(fmt.Sprintf("SELECT id, type, name, slug, created_at, updated_at FROM %s WHERE slug = ?;", ssoProviderTableName), slug).Scan(&provider); tx.RowsAffected == 0 {
```

Comment:
We're almost always going to want to include the SSO provider details. Consider using a `left join` and set the appropriate values in the returned `model.SSOProvider`

Reply:
I hit a wall trying to resolve the SQL in combination with gorm's scan. Opted to just use preload 🤷